lib/components/helpers/array-item-control.js

```diff
@@ -44,12 +44,16 @@ export default React.createClass({
       readOnly: isLastItem && !config.isRemovalOfLastArrayItemAllowed(field)
     });
 
-    const moveItemForward = this.props.index < (this.props.numItems - 1) ?
-      config.createElement('move-item-forward', { field: field, onClick: this.onMoveForward }) :
+    const moveItemForward = field.fieldIndex !== field.parent.value.length - 1 ?
+      config.createElement('move-item-forward', {
+        field: field, onClick: this.onMoveForward
+      }) :
       null;
 
-    const moveItemBack = this.props.index > 0 ?
-      config.createElement('move-item-back', { field: field, onClick: this.onMoveBack }) :
+    const moveItemBack = field.fieldIndex !== 0 ?
```

Comment:
Same question here: `field.fieldIndex` should be the same as `this.props.index` so I'm not seeing how this changes anything. Maybe I'm mistaken though, or maybe there's a bug with `index`?